### DISCUSSION TITLE
How do I handle private pypi with pep621 manager ?

### DISCUSSION THREAD

**Discussion by clementfumey** [renovatebot/renovate#24436] [Request Help] [answered]
### How are you running Renovate?

Self-Hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.93.7 docker image renovate/renovate:36-full

### If you're self-hosting Renovate, select which platform you are using.

bitbucket-server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I was expecting pep621 manager to retrieve package dependencies of my pyproject.toml set in an external pypi.

Here is my [project]
```
dependencies = [
  "crcmod==1.7",
  "pycryptodome==3.11.0",
  "myownpackage @ https://nexus.myorganisation.com/repository/pypi/packages/myownpackage/1.0.15/myownpackage-1.0.15-py3-none-any.whl",
]
```

Running pip install on this pyproject.toml work and pip will fetch myownpackage in the right place. Within renovate i have the following error message

`DEBUG: Dependency myownpackage  has unsupported/unversioned value @https://nexus.myorganisation.com/repository/pypi/packages/myownpackage/1.0.15/myownpackage-1.0.15-py3-none-any.whl (versioning=pep440) (repository=ZIGTST/testrenovate, baseBranch=test-dependencies)`

### What you tried so far.

I have tried to add a checksum at the end of the pypi private url as it is said in [pep440](https://peps.python.org/pep-0440/#direct-references)

```
dependencies = [
  "crcmod==1.7",
  "pycryptodome==3.11.0",
  "myownpackage @ https://nexus.myorganisation.com/repository/pypi/packages/myownpackage/1.0.15/myownpackage-1.0.15-py3-none-any.whl#sha256=3deaa92009d82c13d55ac98950f853a65b119b292fefc5269759a71346652733",
]
```

but I still have the same error in the log

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Dependency myownpackage  has unsupported/unversioned value @https://nexus.myorganisation.com/repository/pypi/packages/myownpackage/1.0.15/myownpackage-1.0.15-py3-none-any.whl (versioning=pep440) (repository=ZIGTST/testrenovate, baseBranch=test-dependencies)
```

</details>

**Comment by secustor** [comment, marked as answer]
`pep621` does not support inline references, tough you can use the `regex` manager to support this. 

Alternatively you can use the private registry support for package managers like PDM or peotry.
